schema/snapcraft.yaml

```diff
@@ -324,13 +324,27 @@ properties:
         special case, 'plugins' is also not a valid part name."
     patternProperties:
       ^(?!plugins$)[a-z0-9][a-z0-9+-\/]*$:
-        # Make sure snap/prime are mutually exclusive
         allOf:
+          # Make sure snap/prime are mutually exclusive
           - not:
               type: object
               required: [snap, prime]
               validation-failure:
                   "{.instance} cannot contain both 'snap' and 'prime' keywords."
+          # Make sure prepare/pre-build are mutually exclusive
+          - not:
+              type: object
+              required: [prepare, pre-build]
+              validation-failure:
+                  "{.instance} cannot contain both 'prepare' and 'pre-build'
+                  keywords. Use 'pre-build'."
+          # Make sure install/post-build are mutually exclusive
+          - not:
+              type: object
+              required: [install, post-build]
```

Comment:
we should make sure that `pre-build` and `install` do not show up together either

Reply:
and viceversa